src/libasr/codegen/asr_to_c.cpp

```diff
@@ -634,6 +635,20 @@ R"(
         src = out;
     }
 
+    void visit_StringItem(const ASR::StringItem_t& x) {
+        this->visit_expr(*x.m_idx);
+        std::string idx = std::move(src);
+        this->visit_expr(*x.m_arg);
+        std::string str = std::move(src);
+        src = "(char *)memcpy((char *)calloc(2U, sizeof(char)), "
+                + str + " + " + idx + " - 1, 1U)";
```

Comment:
For the `C` Backend, currently, I am allocating memory into heap using `calloc`. Ideally, I guess, the same memory should be freed. I tried freeing it [here](https://github.com/lcompilers/lpython/pull/781#discussion_r928645818).